src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -346,7 +347,7 @@ Handle java_lang_String::create_from_str(const char* utf8_str, TRAPS) {
 #ifdef ASSERT
   // This check is too strict when the input string is not a valid UTF8.
   // For example, it may be created with arbitrary content via jni_NewStringUTF.
-  if (UTF8::is_legal_utf8((const unsigned char*)utf8_str, (int)strlen(utf8_str), false)) {
+  if (UTF8::is_legal_utf8((const unsigned char*)utf8_str, strlen(utf8_str), false)) {
```

Comment:
Most of the time we use `is_legal_utf8`, we have a char* and have to cast it to unsigned char*.  How about adding an inlined overload for is_legal_utf8(const char*, size_t) for conenience?

Reply:
Sorry out of scope for this change.